api/errors.go

```diff
@@ -1,47 +1,118 @@
 package api
 
 import (
+	"context"
 	"fmt"
 	"net/http"
+	"os"
+	"runtime/debug"
 )
 
-func badRequestError(w http.ResponseWriter, fmtString string, args ...interface{}) *HTTPError {
-	err := httpError(http.StatusBadRequest, fmtString, args...)
-	sendJSON(w, err.Code, err)
-	return err
+var genericInternalServerErrorResponse = []byte(`{"code":500,"msg":"Internal server error"}`)
+
+func badRequestError(fmtString string, args ...interface{}) *HTTPError {
+	return httpError(http.StatusBadRequest, fmtString, args...)
 }
 
-func internalServerError(w http.ResponseWriter, fmtString string, args ...interface{}) *HTTPError {
-	err := httpError(http.StatusInternalServerError, fmtString, args...)
-	sendJSON(w, err.Code, err)
-	return err
+func internalServerError(fmtString string, args ...interface{}) *HTTPError {
```

Comment:
You mean return the existing `request_id` in the JSON response? I was thinking maybe we do that as a *header* for all requests anyways?